phoenix-core-client/src/main/java/org/apache/phoenix/optimize/QueryOptimizer.java

```diff
@@ -247,6 +248,10 @@ private List<QueryPlan> getApplicablePlansForSingleFlatQuery(QueryPlan dataPlan,
             }
             plans.add(0, hintedPlan);
         }
+        if (dataPlan.getContext().getScanRanges().isPointLookup()
+                && stopAtBestPlan && dataPlan.isApplicable()) {
+            return Collections.<QueryPlan> singletonList(dataPlan);
+        }
```

Comment:
Why is this logic added here ? The dataPlan is already added to the list of plans. This feels like code smell.

Reply:
I was thinking about the same.

It seems that we might have more plans in the list of plans than just data plan, so if we return single plan here as data plan, it would help cover this case: Even if the hinted index is not applicable, and the data table has point lookup, we return best plan as point lookup on data table.

Reply:
Yes, this fix honors index hint even if original data plan is point lookup. But if hinted plan is null or not applicable then we need to return original data plan as we are already maximum optimized.

Reply:
The optimizer will select the optimal plan from the list of plans. You can test by commenting this logic. There is no need to duplicate code. It is confusing to maintain. Same logic is in multiple places.

Reply:
Got it, thanks. Checking

Reply:
@tkhurana I see that if we remove this block then code goes on to create query plan for each index of the original data table. And, to create query plan for each index at least 1 RPC call is done so, I think this can be expensive when the end result we know is original data plan but we will still do RPC calls for other indexes.

Reply:
But this will not be a common scenario. Most of the time index hint will be null and you will return early. If index hint is not null then most of the times you will use the hinted index plan. It will be rare that you will end up here that you have to exit early.

Reply:
I have addressed the suggestion and also added an IT for partial index. Please take a look. Thanks